src/rdkafka_mock_handlers.c

```diff
@@ -2488,6 +2519,237 @@ rd_kafka_mock_handle_OffsetForLeaderEpoch(rd_kafka_mock_connection_t *mconn,
+
+        /* RackId */
+        rd_kafka_buf_read_str(rkbuf, &RackId);
+
+        /* RebalanceTimeoutMs */
+        rd_kafka_buf_read_i32(rkbuf, &RebalanceTimeoutMs);
+
+        /* #SubscribedTopicNames */
+        rd_kafka_buf_read_arraycnt(rkbuf, &SubscribedTopicNamesCnt,
+                                   RD_KAFKAP_TOPICS_MAX);
+        if (SubscribedTopicNamesCnt >= 0) {
+                SubscribedTopicNames = rd_calloc(
+                    SubscribedTopicNamesCnt > 0 ? SubscribedTopicNamesCnt : 1,
+                    sizeof(rd_kafkap_str_t));
+                for (i = 0; i < SubscribedTopicNamesCnt; i++) {
+                        /* SubscribedTopicNames[i] */
+                        rd_kafka_buf_read_str(rkbuf, &SubscribedTopicNames[i]);
+                }
+        }
+
+        /* ServerAssignor */
+        rd_kafka_buf_read_str(rkbuf, &ServerAssignor);
+
+        /* #TopicPartitions */
+        rd_kafka_buf_read_arraycnt(rkbuf, &TopicPartitionsCnt,
+                                   RD_KAFKAP_PARTITIONS_MAX);
+        if (TopicPartitionsCnt)
+                current_assignment =
+                    rd_kafka_topic_partition_list_new(TopicPartitionsCnt);
+        for (i = 0; i < TopicPartitionsCnt; i++) {
+                int32_t PartitionsCnt;
+                rd_kafka_Uuid_t TopicId;
+                rd_kafka_mock_topic_t *mtopic;
+                int j;
+
+                /* TopicId */
+                rd_kafka_buf_read_uuid(rkbuf, &TopicId);
+
+                mtopic = rd_kafka_mock_topic_find_by_id(mcluster, TopicId);
+
+                /* #Partitions */
+                rd_kafka_buf_read_arraycnt(rkbuf, &PartitionsCnt,
+                                           RD_KAFKAP_PARTITIONS_MAX);
+                for (j = 0; j < PartitionsCnt; j++) {
+                        int32_t Partition;
+                        rd_kafka_topic_partition_t *rktpar;
+
+                        /* Partitions[j] */
+                        rd_kafka_buf_read_i32(rkbuf, &Partition);
+
+                        if (mtopic) {
+                                rktpar = rd_kafka_topic_partition_list_add(
+                                    current_assignment, "", Partition);
+                                rd_kafka_topic_partition_set_topic_id(rktpar,
+                                                                      TopicId);
+                        }
+                }
+
+                rd_kafka_buf_skip_tags(rkbuf);
+        }
+
+        /* Inject error, if any */
+        err = rd_kafka_mock_next_request_error(mconn, resp);
+
+        if (!err) {
+                mrkb = rd_kafka_mock_cluster_get_coord(
+                    mcluster, RD_KAFKA_COORD_GROUP, &GroupId);
+
+                if (!mrkb)
+                        err = RD_KAFKA_RESP_ERR_COORDINATOR_NOT_AVAILABLE;
+                else if (mrkb != mconn->broker)
+                        err = RD_KAFKA_RESP_ERR_NOT_COORDINATOR;
+        }
+
+        if (!err) {
```

Comment:
There two can be merged.

Reply:
No, because if the coordinator isn't available it doesn't need have to add the member to the group.